File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        
        int i=0;
        int n= numbers.size();
        int j = n-1;

        while(i<j)
        {
            int sum = numbers[i] + numbers[j];

            if(sum==target)
                return {i+1,j+1};
            
            if(sum>target)
                j--;
            
            if(sum<target)
                i++;
        }

        return {1,1};


    }
};

File: Data Structures & Algorithms/binary-search/submission-0.ts
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number}
     */
    search(nums: number[], target: number): number {

        let start:number = 0
        let end:number = nums.length -1

        while(start<=end){

            let mid:number = Math.floor(start + (end - start) / 2)

            if(nums[mid] === target){
                return mid
            }
            if(nums[mid] < target){
                start = mid+1
            }
            if(nums[mid] > target){
                end = mid-1
            }
        }
        return -1
    }
}
